### PR TITLE
ci: speed up docker image build

### PR DESCRIPTION
**Pull request by Zxilly**
1. use the go compiler to cross compile but not the qemu emulator as it's too slow.
2. always attach a `master` tag to the image point to the latest commit in repo.

### CODE REVIEW

**Comment by funnyzak** [comment]
Tested on ARMv7:

- Running as source or binary, the cache function works fine.
- Packaged image locally, tested and found to be problematic. **Troubleshooting may be related to creating containers using `scratch`**.

When running as a container, the error message is as follows:
```json

{
	"code": 500,
	"error": "Get \"https://api.github.com/copilot_internal/v2/token\": tls: failed to verify certificate: x509: certificate signed by unknown authority"
}
```

**Comment by Zxilly** [comment]
fixed, timezone file and cert bundle is missing

**Comment by funnyzak** [comment]
I built a multi-platform image and ran it on amd64, arm platforms and everything works fine.

**Review by funnyzak — APPROVED** [review]

**Review by aaamoon — APPROVED** [review]

File: .github/workflows/docker_build.yml [hunk 1]
```diff
@@ -30,8 +30,6 @@ jobs:
         uses: actions/checkout@v3
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v2
-      - name: Set up QEMU
-        uses: docker/setup-qemu-action@v3
       - name: Set up Docker Buildx
```
Comment (funnyzak) [inline .github/workflows/docker_build.yml:33]: It shouldn't be necessary to set up Docker Buildx twice 🤣

**Review by funnyzak — APPROVED** [review]

**Review by Geniucker — APPROVED** [review]

**Pull request merged**